### ISSUE TITLE
HvMessage optimisations

### ISSUE THREAD

**Issue by pingdynasty** [enzienaudio/heavy#72]
Request to change HvMessage.c

```
case HV_MSG_BANG: len[i] = hv_snprintf(NULL, 0, "%s", "bang") + 1; break;
to
case HV_MSG_BANG: len[i] = 5; break;

case HV_MSG_SYMBOL: len[i] = hv_snprintf(NULL, 0, "%s", msg_getSymbol(m, i)) + 1; break
to
case HV_MSG_SYMBOL: len[i] = hv_strlen(msg_getSymbol(m, i)) + 1; break

and suitable changes to use strncpy() instead of snprintf() in
case HV_MSG_BANG: hv_snprintf(finalString+pos, len[i], "%s", "bang"); break;
and
case HV_MSG_SYMBOL: hv_snprintf(finalString+pos, len[i], "%s", msg_getSymbol(m, i)); break;
```

**Comment by mhroth** [comment]
@pingdynasty does the OWL not like `hv_snprintf`?

**Comment by pingdynasty** [comment]
printf-family are quite weighty, would be good to loose a bit of code
size and ram: about 3k.
I might substitute for my own ftoa() and itoa() functions, try it out
and send a diff.
Am I right in thinking `msg_toString()` is only called when the print
hook is used?

**Comment by mhroth** [comment]
Yes, that's correct. Heavy only calls `msg_toString` for the `print` object. The function is exposed in the top-level API as well, so potentially others are using it too. I will make your changes though, they're a good idea. You wouldn't believe how old that code is even if I told you :)

And if you want to send over at least a suitable implementation of `ftoa` I could probably use that too.

**Comment by pingdynasty** [comment]
Now I think there's a typo in HvMessage.c:

```
./Build/Source/HvMessage.c:250:25: error: expected ')' before '!' token
   hv_assert(finalString ! = NULL);
```

**Comment by pingdynasty** [comment]
Here's my HvMessage.c:
https://github.com/pingdynasty/OwlProgram/blob/memmathperf/HeavySource/HvMessage.c
And the ftoa/itoa implementations:
https://github.com/pingdynasty/OwlProgram/blob/memmathperf/Source/message.cpp

**Comment by mhroth** [comment]
!! `hv_assert(finalString ! = NULL);` is fixed now.
